yb-voyager/src/srcdb/postgres.go

```diff
@@ -1153,9 +1154,79 @@ func (pg *PostgreSQL) GetMissingAssessMigrationPermissions() ([]string, error) {
 		combinedResult = append(combinedResult, fmt.Sprintf("\n%s[%s]", color.RedString("Missing SELECT permission for user %s on Tables: ", pg.source.User), strings.Join(missingTables, ", ")))
 	}
 
+	result, err := pg.checkPgStatStatementsSetup()
+	if err != nil {
+		return nil, fmt.Errorf("error checking pg_stat_statement extension installed with read permissions: %w", err)
+	}
+
+	if result != "" {
+		combinedResult = append(combinedResult, result)
+	}
 	return combinedResult, nil
 }
 
+const (
+	queryPgStatStatementsSchema = `
+		SELECT nspname
+		FROM pg_extension e
+		JOIN pg_namespace n ON e.extnamespace = n.oid
+		WHERE e.extname = 'pg_stat_statements'`
+
+	querySharedPreloadLibraries = `SELECT current_setting('shared_preload_libraries')`
+
+	queryHasReadStatsPermission = `
+		SELECT pg_has_role(current_user, 'pg_read_all_stats', 'USAGE')`
+)
+
+// checkPgStatStatementsSetup checks if pg_stat_statements is properly installed and if the user has the necessary read permissions.
+func (pg *PostgreSQL) checkPgStatStatementsSetup() (string, error) {
+	if !utils.GetEnvAsBool("REPORT_UNSUPPORTED_QUERY_CONSTRUCTS", true) {
+		log.Infof("REPORT_UNSUPPORTED_QUERY_CONSTRUCTS is set as false, skipping guardrails check for pg_stat_statements")
+		return "", nil
+	}
+
+	// 1. check if pg_stat_statements extension is available on source
+	var pgssExtSchema string
+	err := pg.db.QueryRow(queryPgStatStatementsSchema).Scan(&pgssExtSchema)
```

Comment:
It can be installed in multiple schemas right? QueryRow will just give us one schema?  Shouldn't we do a difference between userProvidedSchemaList and pgStatStatementsSchemaList?

Reply:
i have tried that, its not possible. We can't create a extension in multiple schemas

```
mig_assessment=# \dx
                                            List of installed extensions
        Name        | Version |   Schema   |                              Description
--------------------+---------+------------+------------------------------------------------------------------------
 pg_stat_statements | 1.9     | public     | track planning and execution statistics of all SQL statements executed
 pgcrypto           | 1.3     | public     | cryptographic functions
 plpgsql            | 1.0     | pg_catalog | PL/pgSQL procedural language
 postgis            | 3.5.0   | public     | PostGIS geometry and geography spatial types and functions
(4 rows)

mig_assessment=# set search_path to ext;
SET
mig_assessment=# create extension pg-
mig_assessment=# create extension pg_stat_statements ;
ERROR:  extension "pg_stat_statements" already exists
mig_assessment=# \dx
                                            List of installed extensions
        Name        | Version |   Schema   |                              Description
--------------------+---------+------------+------------------------------------------------------------------------
 pg_stat_statements | 1.9     | public     | track planning and execution statistics of all SQL statements executed
 pg_surgery         | 1.0     | ext        | extension to perform surgery on a damaged relation
 pgcrypto           | 1.3     | public     | cryptographic functions
 plpgsql            | 1.0     | pg_catalog | PL/pgSQL procedural language
 postgis            | 3.5.0   | public     | PostGIS geometry and geography spatial types and functions
(5 rows)

``

Reply:
I don't think an extension can be created in more than one schema as per [this](https://www.postgresql.org/docs/current/sql-createextension.html#:~:text=Remember%20that%20the%20extension%20itself%20is%20not%20considered%20to%20be%20within%20any%20schema%3A%20extensions%20have%20unqualified%20names%20that%20must%20be%20unique%20database%2Dwide.%20But%20objects%20belonging%20to%20the%20extension%20can%20be%20within%20schemas.)

Reply:
We are enforcing user to export the schema which has pgss extension, which is not correcet.
We should just warn/prompt the user if pgss ext schema is different from export schema_list.

decide to take this in a separate ticket. cc  @makalaaneesh

Reply:
Guardrail is a prompt and I think we are just warning in the metadata script so where are we enforcing pgss extension's to be in schema_list?

Reply:
@priyanshi-yb  similar check is there in pg gather metadata script.
So if user doesn't specify the pgss_ext_schema in schema_list then we won't detect/report UQC